modules/models.py

```diff
@@ -276,6 +277,24 @@ def ctransformers_loader(model_name):
     model, tokenizer = ctrans.from_pretrained(model_file)
     return model, tokenizer
 
+def AutoAWQ_loader(model_name):
+   from awq import AutoAWQForCausalLM
+
+   model_dir = Path(f'{shared.args.model_dir}/{model_name}')
+
+   if shared.args.deepspeed:
+       logger.warn("AutoAWQ is incompatible with deepspeed")
+
+   model = AutoAWQForCausalLM.from_quantized(
+       quant_path=model_dir,
+       max_new_tokens=shared.args.max_seq_len,
```

Comment:
Does it make sense to use max_seq_len for max_new_tokens? For some models, this parameter is automatically set to 16384 or 32768 based on the data in the config.json file for the model. 

Is "max_new_tokens" meant to represent the entire context length just like in exllama or is it a cache for (literally) the new tokens? If the latter, max_new_tokens shouldn't be reutilized for this.

Reply:
Max new tokens is used universally for setting the context length and cache size. Cache size is only in use when using fused modules as the rest is controlled by transformers/accelerate.

Reply:
I see, but still, there seems to be an over-allocation of memory somewhere. AutoAWQ is using 1.2GiB more VRAM than a similar GPTQ model:

8011MiB
python server.py --loader AutoAWQ --model TheBloke_Llama-2-7B-AWQ

6759MiB
python server.py --loader exllama --model TheBloke_Llama-2-7B-GPTQ_gptq-4bit-128g-actorder_True

In both cases, max_seq_len is set to 4096.

Reply:
@oobabooga Thanks for bringing this up. I just fixed this behavior on main and it saves ~2 GB VRAM.

Using my own benchmarking, this is before and after the latest commit:
Before: 8.11 GB, After: 6.17 GB

Reply:
Nice! I'm glad that this was a bug that could be fixed instead of some more fundamental issue with AutoAWQ.